### PR TITLE
Feature/distributed

### PR DESCRIPTION
**Pull request by jrobinAV**
- Refactor Scheduler and dispatcher so we better-split responsibilities : 
    * The Scheduler only does the scheduling. It manages the job statuses. It instantiates a Dispatcher.
    * The dispatcher dispatch jobs and execute callbacks. It does not participate to scheduling anymore. The Dispatcher has been cut into three classes. 
        - `JobDispatcher` contains functions that do not depend on the implementation. 
        - `StandaloneJobDispatcher(JobDispatcher)` is a future based implementation for a // and asynchronous execution.
        - `DevelopmentJobDispatcher(StandaloneJobDispatcher)` is future based as well but for synchronous execution.
 - Prepare the code for a StandaloneDispatcher implementation in taipy-enterprise

### CODE REVIEW

File: taipy/core/_scheduler/_dispatcher/_development_job_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from taipy.core._scheduler._dispatcher._standalone_job_dispatcher import _StandaloneJobDispatcher
+from taipy.core._scheduler._executor._synchronous import _Synchronous
+
+
+class _DevelopmentJobDispatcher(_StandaloneJobDispatcher):
+    """Manages job dispatching (instances of `Job^` class) in a synchronous way."""
+
+    def __init__(self):
+        super().__init__()
+        self._executor = _Synchronous()
```
Comment (florian-vuillemot) [inline taipy/core/_scheduler/_dispatcher/_development_job_dispatcher.py:21]: The Synchronous executor is used here only. Can we move it here?

Comment (jrobinAV) [reply taipy/core/_scheduler/_dispatcher/_development_job_dispatcher.py:21]: What do you mean? You want to put the Synchronous code directly in the _DevelopmentJobDispatcher class ?

Comment (florian-vuillemot) [reply taipy/core/_scheduler/_dispatcher/_development_job_dispatcher.py:21]: yes

**Review by toan-quach — APPROVED** [review]
LGTM!

**Pull request merged**